### ISSUE TITLE
Can't create new Zone for Website

### ISSUE THREAD

**Issue by sgreiner** [revive-adserver/revive-adserver#1134]
I want to create a new Zone for one of our websites.
But it fails with the following message:

> An error occurred while accessing the database
> Due to a problem with the database Revive Adserver couldn't retrieve or store data. If this problem is reproducable it might be caused by a bug in Revive Adserver. Please report the following information to the creators of Revive Adserver. Also try to describe the actions that led to this error as clearly as possible.

Version:  Revive Adserver v5.0.2
PHP: PHP 7.2.24
MySQL: 5.7.27

After copying the failing Query to my DB console: I see the reason:

> INSERT INTO ox_zones
(affiliateid , zonename , description , delivery , zonetype , category , width , height , ad_selection , chain , prepend , append , comments , updated , oac_category_id , show_capped_no_cookie )
VALUES
( 35 , 'XXXX' , '' ,  0 ,  3 , '' ,  468 ,  60 , '' , '' , '' , '' , '' , '2019-12-03 14:46:45' ,  NULL ,  0 );

`Error Code: 1364. Field 'what' doesn't have a default value`

The "what" field is defined as MEDIUMTEXT and NOT NULL.

If the create zone dialog is not sending a "what" content - then the field should either be not defined as "NOT NULL" or it should have a default value.

**Comment by sgreiner** [comment]
Simply setting the zones.what field to default '' (nothing) does not work:
Error Code: 1101. BLOB, TEXT, GEOMETRY or JSON column 'what' can't have a default value

So I guess it must not be "NOT NULL" but "NULL"

Is that save to change?

**Comment by erikgeurts** [comment]
Our Github pages are not intended for technical support. Please consider posting your question on our community forums at https://forum.revive-adserver.com. If it turns out this is a bug or development issue after all, we'd be happy to have a closer look.

**Comment by sgreiner** [comment]
Excuse me, but I don't want technical support - I want to report a bug.

It is an incompatibility between revive-adserver and a MySQL 5.7 and 8 server where an essential feature like creating a zone is not working, which prevents working with the product at all.

So can we take care of the breaking bug here or should I create a forum post beforehand?

**Comment by mbeccati** [comment]
I could bet on the fact that you don't have `mysql4_compatibility=1` in the database section of your configuration.

**Comment by sgreiner** [comment]
@mbeccati yes you are totally right. But we had to remove it unfortunately.
With mysql4_compatibility=1 we got an error because we can't use this on MySQL 8.x :-(
But we needed to migrate (for our test servers, and soon production system). Thank you for finding this. But my problem persists unfortunately.

[MySQL 5.7.x available SQL modes](https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sqlmode_mysql40)
[MySQL 8.x available SQL modes]https://dev.mysql.com/doc/refman/8.0/en/sql-mode.html

But I guess, I can close this issue right now.

I checked https://www.revive-adserver.com/support/requirements/ and see that only MySQL 4 is currently fully supportet. I am sorry for creating the issue, should have checked that page first.

> Database server requirements:
> MySQL 4.1+, or
> PostgreSQL 8.1+

> At the time of this page being updated, MySQL 8.x is NOT supported for use with any version of Revive Adserver, including the upcoming v5.0.0 (scheduled for September 2019).
> When using the ‘utf8mb4’ character set, MySLQ 5.x is not supported either

Thank you guys anyways.
